tools/bin/mtest

```diff
@@ -238,115 +239,149 @@ def toolchain_exes(target):
 
 def toolchain(target, exit_on_failure=True):
     # basic CFLAGS setup
-    c_flags = f"-I{ROOT}/include"
-    if "-O" not in target.cflags:
-        if args.size_tfm and "clang" in target.compiler:
-            c_flags += " -Oz"
-        else:
-            c_flags += " -Os"
-    if "-std" not in target.cflags: c_flags += " -std=c99"
+    # target_c_flags may get overridden by the user (and dropped from the resulting set of flags);
+    # c_flags is always applied
+
+    target_c_flags = ["-std=c99", f"-I{ROOT}/include"]
+
+    if args.size_tfm and "clang" in target.compiler:
```

Comment:
Does this match `armclang`?

Reply:
Should it use `compiler_family`?

Reply:
Yes

Reply:
No benefit to changing to compiler_family: this is maybe poorly named - it means "compiler without version", so "clang" rather than "clang-15", but "armclang" is separate.